### ISSUE TITLE
`cupy.vectorize` does not work in the interactive mode 

### ISSUE THREAD

**Issue by leofang** [cupy/cupy#4431]
I was looking into how `inspect` and `ast` work together, and I noticed that a function object defined in an interactive interpreter session would not contain the code information, so `inspect.getsource(func)` would emit an `OSError`. As a result, while the following snippet works as expected in the script mode:
```python
import cupy as cp

def f(x, y):
    return x + y

vec_f = cp.vectorize(f)
a = cp.random.random(10)
b = cp.random.random(10)
c = vec_f(a, b)
assert cp.allclose(c, a+b)
```
an error is raised in the interactive mode:
```python
>>> import cupy as cp
>>> 
>>> def f(x, y):
...     return x + y
... 
>>> 
>>> vec_f = cp.vectorize(f)
>>> a = cp.random.random(10)
>>> b = cp.random.random(10)
>>> c = vec_f(a, b)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/leofang/dev/cupy_cuda10.2/cupy/_functional/vectorize.py", line 68, in __call__
    code, ret_type = func._emit_code_from_types(in_types, ret_type)
  File "/home/leofang/dev/cupy_cuda10.2/cupyx/jit/_interface.py", line 28, in _emit_code_from_types
    self.func, self.attributes, self.mode, in_types, ret_type)
  File "/home/leofang/dev/cupy_cuda10.2/cupyx/jit/_compile.py", line 27, in transpile
    source = inspect.getsource(func)
  File "/home/leofang/miniconda3/envs/cupy_cuda102_dev/lib/python3.7/inspect.py", line 973, in getsource
    lines, lnum = getsourcelines(object)
  File "/home/leofang/miniconda3/envs/cupy_cuda102_dev/lib/python3.7/inspect.py", line 955, in getsourcelines
    lines, lnum = findsource(object)
  File "/home/leofang/miniconda3/envs/cupy_cuda102_dev/lib/python3.7/inspect.py", line 786, in findsource
    raise OSError('could not get source code')
OSError: could not get source code
```
It's probably not a big deal as most people would run in the script mode, but may worth keeping track.

**Comment by leofang** [comment]
Ref: https://bugs.python.org/issue12920

**Comment by emcastillo** [comment]
yeah, it's the drawback of using AST instead of bytecode :(

**Comment by leofang** [comment]
Fixed by #5012.
